feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/domain/validation/SwapValidations.kt

```diff
@@ -72,6 +94,16 @@ fun SwapValidationSystemBuilder.sufficientAssetOutBalanceToStayAboveED(
     assetSourceRegistry: AssetSourceRegistry
 ) = sufficientAmountOutToStayAboveEDValidation(assetSourceRegistry)
 
+fun SwapValidationSystemBuilder.sufficientBalanceToPayFeeConsideringED(
+    assetSourceRegistry: AssetSourceRegistry
+) = enoughBalanceToStayAboveEDValidation(
+    assetSourceRegistry,
+    fee = { it.feeAsset.token.amountFromPlanks(it.swapFee.networkFee.amount) },
+    balance = { it.feeAsset.free },
```

Comment:
Verify with Ruslan about `free` or `total`

Reply:
I asked Ruslan yesterday and he approved that we use `free` balance